utils/docker/peer/Dockerfile

```diff
@@ -0,0 +1,100 @@
+# Copyright IBM Corp. All Rights Reserved.
```

Comment:
This is more or less a true super-set of dev/Dockerfile
-> 
Couldn't you just build on top of that image?  E.g., "backport" all changes you have done up except make plugin and last three ENV/CMD statement to the dev image and then build the peer image from the dev image, just with the plugin build and ENV/CMDs in the peer image as commands?

Reply:
I have made this change locally. The only concern I have before I update the PR is that this Peer image always copies in the entire docker context, which essentially means they every time the dev image is built it will not be able to use the cache for FPC repo images layers. I am thinking that is all right because chances are if you are rebuilding there are changes in the FPC repo. What do you think?

Reply:
As long as you run in the "default" mode from within the dev/peer directory, it shouldn't really be a problem as it would only be an issue of the Dockerfile has changed? Or is docker not smart enough to realize that? There are also interesting workaround to make copy conditional (e.g., ONBUILD variant from https://stackoverflow.com/questions/31528384/conditional-copy-add-in-dockerfile) but they are still rather kludgy and not worth adopting.

I guess there is the other potential issue that you might want peer and dev image with different repos (e.g., peer from local repo whereas you might have a peer with the remote repo) but that could be solved simply by naming conventions of the user.  

So maybe you are right and having duplicates is better. My main concern was inconsistent updates. But hopefully we don't have to update much.  What might help though is adding some comments in the Dockerfiles themselves pointing out the duplication and the potential need of replicating changes also to the "sibling"?

Reply:
Unfortunately, docker doesn't know it has the same stuff until it builds the layer again. I did look into the conditional building but didn't like the approaches there.

One approach is having one more intermediary Dockerfile which takes all the common pieces and then dev will only have the FPC clone piece and the peer will have its varied FPC clone. Would that be more desirable? We could also make some changes to the images so we have a ccenv file, a base that builds from ccenv, and a dev that builds from base, and a peer that builds from base.

Reply:
I guess let's just leave it for now and revisit if ever we run into issues with keeping the similar files in sync. I thought there might be an easy clean way to reuse but evidently this is not the case. So doesn't seem worth spending more time on this now.